overlord/devicestate/firstboot.go

```diff
@@ -75,6 +75,49 @@ func criticalTaskEdges(ts *state.TaskSet) (beginEdge, beforeHooksEdge, hooksEdge
 	return beginEdge, beforeHooksEdge, hooksEdge, nil
 }
 
+// maybeEnforceValidationSetsTask returns a task for tracking validation-sets. This may
+// return nil if no validation-sets are present.
+func maybeEnforceValidationSetsTask(st *state.State, model *asserts.Model, db asserts.RODatabase) (*state.Task, error) {
+	vsKey := func(accountID, name string) string {
+		return fmt.Sprintf("%s/%s", accountID, name)
+	}
+
+	// Encode validation-sets included in the seed
+	as, err := db.FindMany(asserts.ValidationSetType, nil)
+	if err != nil {
+		// If none are included, then skip this
+		if errors.Is(err, &asserts.NotFoundError{}) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	vsas := make(map[string][]byte)
+	for _, a := range as {
+		vsa := a.(*asserts.ValidationSet)
+		vsas[vsKey(vsa.AccountID(), vsa.Name())] = asserts.Encode(a)
+	}
+
+	// Set up pins from the model
+	pins := make(map[string]int)
+	for _, vs := range model.ValidationSets() {
+		key := vsKey(vs.AccountID, vs.Name)
+		if _, ok := vsas[key]; !ok {
+			return nil, fmt.Errorf("missing validation set assertion for %q in the seed", key)
+		}
+		if vs.Sequence > 0 {
+			pins[key] = vs.Sequence
+		}
+	}
+
+	t := st.NewTask("enforce-validation-sets", i18n.G("Track validation sets"))
```

Comment:
So instead of creating a new taks, I did parameterize this one. However I don't know if it should be more structured than a bool ('local') and an int ('userID'). Should I move these into a struct or any other input on this?